src/assembler.jl

```diff
@@ -64,3 +64,122 @@ function assemble!{T}(g::AbstractVector{T}, ge::AbstractVector{T}, edof::Abstrac
         g[edof[i]] += ge[i]
     end
 end
+
+immutable AssemblerSparsityPattern{Tv, Ti}
+    K::SparseMatrixCSC{Tv, Ti}
+    f::Vector{Tv}
+    tmpi::Vector{Int}
+    tmpf::Vector{Tv}
+end
+
+
+function start_assemble(K::SparseMatrixCSC, f::Vector=Float64[])
+    fill!(K.nzval, 0.0)
+    AssemblerSparsityPattern(K, f, Int[], eltype(K)[])
+end
+
+assemble!(A::AssemblerSparsityPattern, Ke::AbstractMatrix, dofs::AbstractVector{Int}) = assemble!(A, eltype(Ke)[], Ke, dofs)
+function assemble!(A::AssemblerSparsityPattern, fe::AbstractVector, Ke::AbstractMatrix, dofs::AbstractVector{Int})
```

Comment:
Ignoring the sorting part, we have that the ratio is `O(nd * nz) / O(nd^2 * log(nz))` = `O(nz / (nd * log(nz)))` so presumable, the threshold should be given in terms of `nz / (nd * log(nz))`